python/lsst/afw/table/base/baseContinued.py

```diff
@@ -0,0 +1,292 @@
+                                (string and Flag) for which views cannot be constructed:
+                                  - 'copy' (default): copy only the unviewable fields.
+                                  - 'raise': raise ValueError if unviewable fields are present.
+                                  - 'skip': do not include unviewable fields in the Astropy Table.
+                                This option is ignored if copy=True.
+        """
+        import astropy.table
+        if cls is None:
+            cls = astropy.table.Table
+        if unviewable not in ("copy", "raise", "skip"):
+            raise ValueError("'unviewable'=%r must be one of 'copy', 'raise', or 'skip'" % (unviewable,))
+        ps = self.getMetadata()
+        meta = ps.toOrderedDict() if ps is not None else None
+        columns = []
+        items = self.schema.extract("*", ordered=True)
+        for name, item in items.items():
+            key = item.key
+            unit = item.field.getUnits() or None  # use None instead of "" when empty
+            if key.getTypeString() == "String":
+                if not copy:
+                    if unviewable == "raise":
+                        raise ValueError("Cannot extract string "
+                                         "unless copy=True or unviewable='copy' or 'skip'.")
+                    elif unviewable == "skip":
+                        continue
+                data = np.zeros(len(self), dtype=np.dtype((str, key.getSize())))
+                for i, record in enumerate(self):
+                    data[i] = record.get(key)
+            elif key.getTypeString() == "Flag":
+                if not copy:
+                    if unviewable == "raise":
+                        raise ValueError("Cannot extract packed bit columns "
+                                         "unless copy=True or unviewable='copy' or 'skip'.")
+                    elif unviewable == "skip":
+                        continue
+                data = self.columns.get_bool_array(key)
+            elif key.getTypeString() == "Angle":
+                data = self.columns.get(key)
+                unit = "radian"
+                if copy:
+                    data = data.copy()
+            else:
+                data = self.columns.get(key)
+                if copy:
+                    data = data.copy()
+            columns.append(
+                astropy.table.Column(
+                    data,
+                    name=name,
+                    unit=unit,
+                    description=item.field.getDoc()
+                )
+            )
+        return cls(columns, meta=meta, copy=False)
+
+    def __getattr__(self, name):
+        # Catalog forwards unknown method calls to its table and column view
+        # for convenience.  (Feature requested by RHL; complaints about magic
+        # should be directed to him.)
+        if name == "_columns":
+            self._columns = None
+            return None
```

Comment:
But I thought we should initialize it to the correct camera view and return that?